src/client/datascience/raw-kernel/liveshare/hostRawNotebookProvider.ts

```diff
@@ -146,11 +145,12 @@ export class HostRawNotebookProvider
         traceInfo(`Getting preferred kernel for ${identity.toString()}`);
         try {
             // We need to locate kernelspec and possible interpreter for this launch based on resource and notebook metadata
-            const kernelConnectionMetadata = await this.kernelSelector.getPreferredKernelForLocalConnection(
+            // TODO: Confirm this logic is valid.
```

Comment:
Some personal TODOs: I need to check if we even need `getPreferredKernelForLocalConnection`
Its possible we need it, but i would like to understand it better (possible its only required for live share, if thats the case, i'll update comments).
FYI - I don't see when the `getPreferredKernelForLocalConnection` will get called for non-live share scenearios - hence the note